### PR TITLE
fix(Filters): various tweaks and fixes

### PR DESCRIPTION
**Pull request by PrincAm**

### CODE REVIEW

File: src/components/Filters/Filters.tsx [hunk 1]
```diff
@@ -35,28 +36,42 @@ const FiltersContainer = styled(FlexContainer)`
...
-  state = getDefaultState(fields),
+  state,
   onApply,
   onChange,
   onClose,
   onCancel,
   isLoading = false,
 }) => {
-  const [filtersValues, setFiltersValues] = useState<Array<Filter>>(state);
+  const [filtersValues, setFiltersValues] = useState<Array<Filter>>(undefined);
```
Comment (ajkl2533) [inline src/components/Filters/Filters.tsx:46]: 'null' would be more appropriate for default state. 'undefined' is more generic and you don't know if it was setup by you or by the app. On the other hand when you use 'null' you are pretty sure it wasn't generated by application

**Review by ajkl2533 — APPROVED** [review]

File: src/components/Filters/Filters.tsx (hunk 1 above)
Comment (PrincAm) [reply src/components/Filters/Filters.tsx:46]: Good point! It is a  good practice, will change it

File: src/components/Filters/Filters.tsx [hunk 2]
```diff
@@ -35,28 +36,42 @@ const FiltersContainer = styled(FlexContainer)`
...
   const [isDefaultState, setIsDefaultState] = useState(true);
   const [hasUnappliedFilters, setHasUnappliedFilters] = useState(false);
 
   useEffect(() => {
```
Comment (youssefssc) [inline src/components/Filters/Filters.tsx:50]: There are 3 useEffects here with shared dependencies. I understand we want to cover all the cases but is it possible to simplify?

Comment (PrincAm) [reply src/components/Filters/Filters.tsx:56]: I understand you feel something suspicious 🙂 There are 3 useStates and each has own useEffect to set the state. If I would merge all useEffects into one with many if elses then some states would be updated even if not necessarily. I believe it is correct and I kind of like how it is separated, for me it is more clear.

Comment (PrincAm) [reply src/components/Filters/Filters.tsx:56]: I've been considering if I shouldn't move it into custom hook

Comment (youssefssc) [reply src/components/Filters/Filters.tsx:56]: My concern is that `fields` is a dependency of the first two useEffects, but I guess fields is a required prop and never changes. If that's correct, then it works :)

**Pull request merged**